src/index.js

```diff
@@ -1,255 +1,161 @@
+      }
+    },
+    autoCleanUri: PropTypes.bool,
+    autoLogin: PropTypes.bool,
+    onLoginFailure: PropTypes.func,
+    onLoginSuccess: PropTypes.func,
     provider: PropTypes.oneOf(config.providers).isRequired,
-    version: PropTypes.string
+    redirect: (props, propName, componentName) => {
+      if (props.provider === 'instagram' && !props[propName] && typeof props[propName] !== 'string') {
+        return new Error(`Missing required \`${propName}\` prop on ${componentName}.`)
+      }
+    }
   }
 
   static defaultProps = {
-    version: '2.8'
+    autoCleanUri: false
   }
 
   constructor (props) {
     super(props)
 
-    this.id = 'sl' + Math.floor(Math.random() * 0xFFFF)
-
-    this.handleSocialLoginInvokeSuccess = this.handleSocialLoginInvokeSuccess.bind(this)
-    this.handleSocialLoginInvokeFailure = this.handleSocialLoginInvokeFailure.bind(this)
-    this.handleLogin = this.handleLogin.bind(this)
-  }
-
-  handleSocialLoginInvokeSuccess (res) {
-    const { callback, provider } = this.props
-
-    //console.log(res)  //uncomment to check response coming from provider in log
+    this.state = {
+      isLoaded: false,
+      isConnected: false,
+      isFetching: false
+    }
 
-    const user = new SocialUser()
-    let userProfile
-    let token
+    // Load required SDK
+    this.sdk = sdk[props.provider]
+    this.accessToken = null
+    this.fetchProvider = props.provider === 'instagram' || props.provider === 'github'
 
-    switch (provider) {
-      case 'google':
-        const profile = window.gapi.auth2.getAuthInstance().currentUser.get().getBasicProfile()
-        const authResponse = window.gapi.auth2.getAuthInstance().currentUser.get().getAuthResponse(true)
+    this.onLoginSuccess = this.onLoginSuccess.bind(this)
+    this.onLoginFailure = this.onLoginFailure.bind(this)
+    this.login = this.login.bind(this)
+  }
 
-        userProfile = {
-          id: profile.getId(),
-          name: profile.getName(),
-          firstName: profile.getGivenName(),
-          lastName: profile.getFamilyName(),
-          email: profile.getEmail(),
-          profilePicURL: profile.getImageUrl()
-        }
-        token = {
-          accessToken: authResponse.access_token,
-          expiresAt: authResponse.expires_at
-        }
+  /**
+   * Loads SDK on componentDidMount and handles auto login.
+   */
+  componentDidMount () {
+    const { appId, appSecret, autoCleanUri, autoLogin, redirect } = this.props
 
-        break
-      case 'facebook':
-        userProfile = {
-          id: res.id,
-          name: res.name,
-          firstName: res.first_name,
-          lastName: res.last_name,
-          email: res.email,
-          profilePicURL: res.picture.data.url
-        }
-        token = {
-          accessToken: res.authResponse.accessToken,
-          expiresAt: res.authResponse.expiresIn
+    this.sdk.load(appId, redirect, appSecret)
+      .then((accessToken) => {
```

Comment:
@carldunham FYI, I updated GitHub code again to use [Gatekeeper project](https://github.com/prose/gatekeeper) instead of the custom implementation I did. It is of course still possible to have your own server implementation while you respect Gatekeeper way of doing (everything is explained clearly in [README](https://github.com/nicolas-goudry/react-social-login/blob/88e0076a82ea853a6b238c0966a992fd817828c1/README.md#github-specifics)).